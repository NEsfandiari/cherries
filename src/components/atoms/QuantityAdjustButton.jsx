import React, { Component } from 'react'
import styled from 'styled-components'

const Container = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 100%;
  color: ${({ color }) => color || ''};
  div {
    cursor: pointer;
  }
  border: 2px solid;
  background-color: white;
  height: ${({ height }) => height || '1.7rem'};
  opacity: ${props => props.opacity};
  transition: 0.3s;
  width: 5rem;
  p {
    margin: 0;
  }
`

class QuantityAdjustButton extends Component {
  state = {}
  render() {
    const { handleAdjust, id, quantity, height, color, opacity } = this.props
    return (
      <Container height={height} color={color} opacity={opacity}>
        <div className="sub" onClick={handleAdjust} data-id={id}>
          -
        </div>
        <p>{quantity}</p>
        <div className="add" onClick={handleAdjust} data-id={id}>
          +
        </div>
      </Container>
    )
  }
}

export default QuantityAdjustButton
