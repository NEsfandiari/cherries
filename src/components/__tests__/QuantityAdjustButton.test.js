import React from 'react'
import renderer from 'react-test-renderer'
import QuantityAdjustButton from '../atoms/QuantityAdjustButton'

describe('QuantityAdjustButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<QuantityAdjustButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
