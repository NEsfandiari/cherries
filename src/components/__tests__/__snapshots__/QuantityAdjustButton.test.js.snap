// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityAdjustButton renders correctly 1`] = `
<div
  className="sc-bdVaJa hBFtSC"
>
  <div
    className="sub"
  >
    -
  </div>
  <p />
  <div
    className="add"
  >
    +
  </div>
</div>
`;
